packages/mask/popups/components/TokenPicker/index.tsx

```diff
@@ -63,35 +81,74 @@ export const TokenPicker = memo(function TokenPicker({
 
     return (
         <Box className={cx(classes.picker, className)} {...rest}>
-            <SelectNetworkSidebar
-                className={classes.sidebar}
-                networks={filteredNetworks}
-                pluginID={NetworkPluginID.PLUGIN_EVM}
-                chainId={sidebarChainId}
-                hideAllButton
-                onChainChange={handleChainChange}
-            />
-            <List className={classes.list} data-hide-scrollbar>
-                {availableAssets.map((asset) => {
-                    const network = filteredNetworks.find((x) => x.chainId === asset.chainId)!
-                    const selected = asset.chainId === chainId && isSameAddress(asset.address, address)
-                    return (
-                        <TokenItem
-                            key={`${asset.chainId}.${asset.address}`}
-                            asset={asset}
-                            network={network}
-                            selected={selected}
-                            onSelect={onSelect}
-                        />
-                    )
-                })}
-            </List>
+            {!lockChainId ?
+                <SelectNetworkSidebar
+                    className={classes.sidebar}
+                    networks={filteredNetworks}
+                    pluginID={NetworkPluginID.PLUGIN_EVM}
+                    chainId={sidebarChainId}
+                    hideAllButton
+                    onChainChange={handleChainChange}
+                />
+            :   null}
+            <FixedSizeList
+                itemCount={availableAssets.length}
+                itemSize={71}
+                height={455}
+                overscanCount={20}
+                itemData={{
+                    tokens: availableAssets,
+                    networks: filteredNetworks,
+                    chainId,
+                    address,
+                    onSelect,
+                }}
+                itemKey={(index, data) => {
+                    const asset = data.tokens[index]
+                    return `${asset.chainId}.${asset.address}`
+                }}
+                style={{
+                    scrollbarWidth: 'none',
+                }}
+                width="100%">
+                {Row}
+            </FixedSizeList>
         </Box>
     )
 })
+
+type RowProps = ListChildComponentProps<{
+    tokens: Array<Web3Helper.FungibleTokenAll | Web3Helper.FungibleAssetAll>
+    networks: Array<ReasonableNetwork<Web3Helper.ChainIdAll, Web3Helper.SchemaTypeAll, Web3Helper.NetworkTypeAll>>
+    chainId?: Web3Helper.ChainIdAll
+    /** Selected address */
+    address?: string
+    onSelect?(asset: Web3Helper.FungibleAssetAll | Web3Helper.FungibleTokenAll): void
+}>
+
+function Row({ data, index, style }: RowProps) {
+    const { tokens, networks, chainId, address, onSelect } = data
+    const asset = tokens[index]
+
+    const network = networks.find((x) => x.chainId === asset.chainId)!
```

Comment:
```suggestion
    const network = networks.find((x) => x.chainId === asset.chainId)
    if (!network) return null;
```

Reply:
`network` is not the critical factor to `TokenItem` but asset is.